File: src/test/java/com/snackgame/server/applegame/fixture/TestFixture.java
package com.snackgame.server.applegame.fixture;

import java.util.List;

import com.snackgame.server.applegame.domain.apple.EmptyApple;
import com.snackgame.server.applegame.domain.apple.PlainApple;
import com.snackgame.server.applegame.domain.game.AppleGameBoard;

public class TestFixture {

    public static final String TWO_BY_FOUR_AS_JSON = "{\"apples\":[[{\"number\":1,\"isGolden\":false},{\"number\":1,\"isGolden\":false},{\"number\":0,\"isGolden\":false},{\"number\":5,\"isGolden\":false}],[{\"number\":9,\"isGolden\":false},{\"number\":2,\"isGolden\":false},{\"number\":0,\"isGolden\":false},{\"number\":2,\"isGolden\":false}]]}";
    public static final String TWO_BY_TWO_WITH_GOLDEN_APPLE_AS_JSON = "{\"apples\":[[{\"number\":1,\"isGolden\":false},{\"number\":8,\"isGolden\":false}],[{\"number\":9,\"isGolden\":true},{\"number\":2,\"isGolden\":false}]]}";

    /**
     * <p>[1, 1, _, 5]</p>
     * <p>[9, 2, _, 2]</p>
     */
    public static AppleGameBoard TWO_BY_FOUR() {
        return new AppleGameBoard(List.of(
                List.of(PlainApple.of(1), PlainApple.of(1), EmptyApple.get(), PlainApple.of(5)),
                List.of(PlainApple.of(9), PlainApple.of(2), EmptyApple.get(), PlainApple.of(2))
        ));
    }

    /**
     * <p>[1, 1]</p>
     * <p>[<b>9</b>, 2]</p>
     * 9는 황금사과이다.
     */
    public static AppleGameBoard TWO_BY_TWO_WITH_GOLDEN_APPLE() {
        return new AppleGameBoard(List.of(
                List.of(PlainApple.of(1), PlainApple.of(8)),
                List.of(PlainApple.of(9).golden(), PlainApple.of(2))
        ));
    }
}

File: src/test/java/com/snackgame/server/game/snackgame/fixture/TestFixture.kt
package com.snackgame.server.game.snackgame.fixture

import com.snackgame.server.game.snackgame.core.domain.Board
import com.snackgame.server.game.snackgame.core.domain.snack.EmptySnack
import com.snackgame.server.game.snackgame.core.domain.snack.GoldenSnack
import com.snackgame.server.game.snackgame.core.domain.snack.PlainSnack

object TestFixture {

    const val TWO_BY_FOUR_AS_JSON =
        "{\"snacks\":[[{\"number\":1,\"isGolden\":false},{\"number\":1,\"isGolden\":false},{\"number\":0,\"isGolden\":false},{\"number\":5,\"isGolden\":false}],[{\"number\":9,\"isGolden\":false},{\"number\":2,\"isGolden\":false},{\"number\":0,\"isGolden\":false},{\"number\":2,\"isGolden\":false}]]}"
    const val TWO_BY_TWO_WITH_GOLDEN_SNACK_AS_JSON =
        "{\"snacks\":[[{\"number\":1,\"isGolden\":false},{\"number\":8,\"isGolden\":false}],[{\"number\":9,\"isGolden\":true},{\"number\":2,\"isGolden\":false}]]}"

    /**
     * 다음과 같은 보드를 생성합니다:
     *
     * `1`, `1`, `_`, `5`
     *
     * `9`, `2`, `_`, `2`
     */
    fun TWO_BY_FOUR(): Board {
        return Board(
            arrayListOf(
                arrayListOf(PlainSnack.of(1), PlainSnack.of(1), EmptySnack.get(), PlainSnack.of(5)),
                arrayListOf(PlainSnack.of(9), PlainSnack.of(2), EmptySnack.get(), PlainSnack.of(2))
            )
        )
    }

    /**
     * 다음과 같은 보드를 생성합니다:
     *
     * `1`, `1`, `_`, `5`
     *
     * `9`, `2`, `_`, `2`
     *
     * `1`, `1`, `3`, `5`
     */
    fun THREE_BY_FOUR(): Board {
        return Board(
            arrayListOf(
                arrayListOf(PlainSnack.of(1), PlainSnack.of(1), EmptySnack.get(), PlainSnack.of(5)),
                arrayListOf(PlainSnack.of(9), PlainSnack.of(2), EmptySnack.get(), PlainSnack.of(2)),
                arrayListOf(PlainSnack.of(1), PlainSnack.of(1), PlainSnack.of(3), PlainSnack.of(5))
            )
        )
    }

    /**
     * 다음과 같은 보드를 생성합니다.
     * 9는 황금 사과입니다
     *
     * `1`, `8`
     *
     * **`(9)`**, `2`
     */
    fun TWO_BY_TWO_WITH_GOLDEN_APPLE(): Board {
        return Board(
            arrayListOf(
                arrayListOf(PlainSnack.of(1), PlainSnack.of(8)),
                arrayListOf(GoldenSnack.of(9), PlainSnack.of(2))
            )
        )
    }
}
